### PR TITLE
Simplify Event

### PR DESCRIPTION
**Pull request by szepeviktor**
I hope it will run!

### CODE REVIEW

File: src/Components/Event.php [hunk 1]
```diff
@@ -13,17 +13,19 @@ class Event extends Component
     /**
      * Create a new component instance.
      *
-     * @return void
+     * @param mixed $listen
      */
-    public function __construct(public string $listen, public string $scope = '{ subscribed, events }')
+    public function __construct(public $listen, public string $scope = '{ subscribed, events }')
```
Comment (Wulfheart) [inline src/Components/Event.php:18]: In my opinionthe function signature should something like this:
`public function __construct(public mixed $listen, public string $scope = '{ subscribed, events }')`

**Review by Wulfheart — CHANGES_REQUESTED** [review]

**Comment by Wulfheart** [comment]
@pascalbaljet Sorry for the audacity to open a review. 😂

File: src/Components/Event.php (hunk 1 above)
Comment (szepeviktor) [reply src/Components/Event.php:18]: Thank you for your comment!

I think `mixed` needs higher version of PHP.
https://github.com/protonemedia/laravel-splade/blob/2f24f7d2140f15393b8d001c4262ccf66efd596e/composer.json#L19

Comment (Wulfheart) [reply src/Components/Event.php:18]: `mixed` has been included in PHP 8 ([rfc](https://wiki.php.net/rfc/mixed_type_v2)).

Comment (szepeviktor) [reply src/Components/Event.php:18]: You are right, Alex.

**Comment by szepeviktor** [comment]
What could be the problem here?

**Comment by szepeviktor** [comment]
Thank you for fixing it.

**Comment by pascalbaljet** [comment]
> I'm not a developer. Could you help me understand why `map('trim')` fails please?

It only trimmed the first element of the collection, not sure why.

**Comment by szepeviktor** [comment]
Arr::map does this: `$items = array_map($callback, $array, $keys);`
`trim`'s second parameter is a list of to-be-trimmed character :(

**Pull request merged**